### PR TITLE
MULE-15378: Be able to generate a Metadata cache ID for a component m…

### PR DESCRIPTION
**Pull request by alegmarra**
…odel based on model parameters

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 1]
```diff
@@ -747,6 +747,7 @@ private void declareConfiguration(ExtensionDeclarer extensionDeclarer, ClassType
         .withOptionalParameter("defaultErrorHandler-ref")
         .ofType(typeLoader.load(String.class))
         .withExpressionSupport(NOT_SUPPORTED)
+        .withAllowedStereotypes(singletonList(ERROR_HANDLER))
```
Comment (gsfernandes) [inline core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java]: Why this change? Is this related to another issue?

File: modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ private void declareMetadata(OperationDeclarer operation, ClassTypeLoader loader
                                                                                 InvokeOutputTypeResolver::new,
                                                                                 NullMetadataResolver::new);
     operation.withModelProperty(new MetadataResolverFactoryModelProperty(() -> factory));
-    operation.withModelProperty(new MetadataKeyIdModelProperty(loader.load(WebServiceTypeKey.class), KEYS_GROUP));
+    operation.withModelProperty(new MetadataKeyIdModelProperty(loader.load(WebServiceTypeKey.class), KEYS_GROUP, "SoapInvoke"));
```
Comment (gsfernandes) [inline modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java]: Is this hardcoded value needed?

Comment (alegmarra) [reply modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java]: it should be a constant (will change), but is indeed needed. That's the name of the Metadata category for the resolver, and in this case there's no need for it to be anonymous.

File: core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java (hunk 1 above)
Comment (alegmarra) [reply core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java]: the allowedStereotypes of a String parameter identify it as a reference to a global element. In this case, the parameter was declared as an errorHandler-ref, but the model said nothing about it.

Comment (alegmarra) [reply core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java]: This 'reference' logic is used here: https://github.com/mulesoft/mule/pull/7110/files#diff-a9272277dbb0602334e99271758985f7R182
